src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/AttackSystem.java

```diff
@@ -127,12 +84,11 @@ public void onInventorySlotChanged(InventorySlotChangedEvent event, EntityRef en
         if (itemIsFlag(item)) {
             String flagTeam = checkWhichFlagPicked(event);
             if (flagTeam.equals(player.getComponent(LASTeamComponent.class).team)) {
-                moveFlagToBase(player, flagTeam);
-                return;
+                player.send(new MoveFlagToBaseEvent(item, flagTeam));
             } else {
                 handleFlagPickup(player, flagTeam);
-                return;
             }
+            return;
```

Comment:
Not sure about my Java skills, but do we really need this for `void` methods? The `return` was previously only used to break early (unnecessarily).